packages/app-builder/src/repositories/TestRunRepository.ts

```diff
@@ -0,0 +1,206 @@
+];
+
+const testrunRuleExecutions: TestRunRuleExecution[] = [
+  {
+    version: 'V1',
+    name: 'Rule 1 name',
+    status: 'hit',
+    total: 10,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V4',
+    name: 'Rule 1 name',
+    status: 'hit',
+    total: 15,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 1 name',
+    status: 'no_hit',
+    total: 20,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V4',
+    name: 'Rule 1 name',
+    status: 'no_hit',
+    total: 15,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 1 name',
+    status: 'error',
+    total: 5,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V4',
+    name: 'Rule 1 name',
+    status: 'snoozed',
+    total: 5,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 2 name',
+    status: 'no_hit',
+    total: 5,
+    rule_id: 'rule-2',
+  },
+  {
+    version: 'V4',
+    name: 'New Rule 2 name',
+    status: 'no_hit',
+    total: 0, // I don't know if this is a possible return from the backend but I'm adding it here to test the UI
+    rule_id: 'rule-2',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 2 name',
+    status: 'hit',
+    total: 15,
+    rule_id: 'rule-2',
+  },
+  {
+    version: 'V4',
+    name: 'New Rule 2 name',
+    status: 'hit',
+    total: 50,
+    rule_id: 'rule-2',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 2 name',
+    status: 'error',
+    total: 15,
+    rule_id: 'rule-2',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 3 name',
+    status: 'hit',
+    total: 15,
+    rule_id: 'rule-3',
+  },
+  {
+    version: 'V4',
+    name: 'Rule 3 name',
+    status: 'hit',
+    total: 15,
+    rule_id: 'rule-3',
+  },
+];
+
+export const makeGetTestRunRepository = () => {
+  return (_: MarbleCoreApi): TestRunRepository => ({
+    getTestRun: ({ testRunId }) => {
+      const run = testruns.find((run) => run.id === testRunId);
+      return run
+        ? Promise.resolve(run)
+        : Promise.reject(new Error('Test run not found'));
+    },
+    listTestRuns: () => Promise.resolve(testruns),
+    listDecisions: () => Promise.resolve(testrunDecisions),
+    listRuleExecutions: () => Promise.resolve(testrunRuleExecutions),
+    launchTestRun: (args: TestRunCreateInput) => {
+      const testRun: TestRun = {
+        id: toUUID(short.generate()),
+        refIterationId: '6f6fe0d8-9a1a-4d5a-bdd7-fa7fcda1b4e3',
+        scenarioId: args.scenarioId,
+        testIterationId: args.testIterationId,
+        startDate: String(new Date().getTime()),
+        endDate: args.endDate,
+        creatorId: '96762987-8895-4af2-9c0a-2dffde09985c',
+        status: (testruns.length === 0
+          ? testRunStatuses[0]
+          : testRunStatuses[
+              randomInteger(1, testRunStatuses.length - 1)
+            ]) as TestRunStatus,
+      };
+
+      testruns.push(testRun);
+
+      return Promise.resolve(testRun);
+    },
+  });
+};
+
+export const makeGetTestRunRepository2 = () => {
```

Comment:
remains to do: plug this ?

Reply:
I'll do this when we'll be ready, since if I do that, I don't have access to my data to test :)